CMake/FindNGHTTP2.cmake

```diff
@@ -21,21 +21,41 @@
 # SPDX-License-Identifier: curl
 #
 ###########################################################################
-include(FindPackageHandleStandardArgs)
 
-find_path(NGHTTP2_INCLUDE_DIR "nghttp2/nghttp2.h")
+if(UNIX)
+  find_package(PkgConfig QUIET)
```

Comment:
not only unix, also windows with vcpkg

Reply:
Thanks, I was wondering about it.

But, I'll handle it in a separate commit, because at the moment almost all such checks are limited to `UNIX`, none of them should be. (or at least nghttp3 shouldn't from where I copied this logic.)

Reply:
MSYS2/mingw-w64 builds might be supporting this too.

Reply:
Or... I might tackle this here and handle the rest separately to avoid a merge queue.

Reply:
Done